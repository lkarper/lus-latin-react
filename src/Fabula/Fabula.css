.Fabula__container {
    width: 75%;
}