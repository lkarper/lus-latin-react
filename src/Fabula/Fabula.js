import React, { Component } from 'react';
import PropTypes from 'prop-types';
import WordsList from '../WordsList/WordsList';
import ENV from '../config';
import './Fabula.css';

class Fabula extends Component {

    static defaultProps = {
        match: {
            params: {
                genre: '',
                id: '',
            },
        },
    }

    state = {
        story: {},
        currentWord: {},
    }

    setCurrentWord = (e, place) => {
        e.preventDefault();
        const { story } = this.state;
        const currentWord = story.vocab[place];
        this.setState({
            currentWord: { [currentWord.word]: currentWord.def },
        });
    }

    componentDidMount() {
        const { genre, id } = this.props.match.params;
        const url = `http://localhost:8000/fabulae?genre=${genre}&id=${id}`;
        const options = { 
            headers: new Headers({
                "Authorization": `Bearer ${ENV.API_TOKEN || '6f06a6cd-90ea-4e28-ab97-9e02ba042203'}`,
            }),
        };
        fetch(url, options)
            .then(res => res.json())
            .then(data => {
                this.setState({
                    story: data,
                });
                console.log(data);
            });
    }

    render() {
        const { story } = this.state;
        if (Object.keys(story).length) {
            const storyLines = story.content.map((line, index) => {
                const wordButtons = line
                    .split(' ')
                    .map((word, i) => {
                        if (Object.keys(story.vocab).includes(`line:${index + 1}_word:${i + 1}`)) {  
                            return (
                                <button 
                                    key={i} 
                                    onClick={(e)=> this.setCurrentWord(e, `line:${index + 1}_word:${i + 1}`)}
                                >
                                    {word}
                                </button>
                            );
                        }
                        return <span key={i}>{` ${word} `}</span>;
                    }); 
                return (
                    <li key={index}>
                        <p>{wordButtons}</p>
                    </li>);
                });
            return (
                <>
                    <article className="Fabula__container">
                        <h3>{story.title}</h3>
                        <ol>
                            {storyLines}
                        </ol>
                    </article>
                    <WordsList currentWord={this.state.currentWord}/>
                </>
            );
        } else {
            return <p className="Fabula__loading">Loading...</p>;
        }
    }
}

Fabula.propTypes = {
    match: PropTypes.shape({
        params: PropTypes.shape({
            genre: PropTypes.string,
            id: PropTypes.string,
        }),
    }),
}

export default Fabula;