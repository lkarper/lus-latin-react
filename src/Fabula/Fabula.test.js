import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Fabula from './Fabula';

describe('Fabula component', () => {
    it('renders the component without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Fabula />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('render the UI as expected', () => {
        const wrapper = shallow(<Fabula />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});