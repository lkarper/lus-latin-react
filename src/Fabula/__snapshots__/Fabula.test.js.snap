// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fabula component renders the UI as expected 1`] = `
<p
  className="Fabula__loading"
>
  Loading...
</p>
`;
